django/db/models/sql/query.py

```diff
@@ -65,12 +68,35 @@ def get_field_names_from_opts(opts):
     )
 
 
+def get_paths_from_expression(expr):
+    if hasattr(expr, "lhs"):
+        yield from get_paths_from_expression(expr.lhs)
+    if hasattr(expr, "rhs"):
+        yield from get_paths_from_expression(expr.rhs)
+    if isinstance(expr, F):
+        yield expr.name
+    if isinstance(expr, Func):
+        for func_expression in expr.source_expressions:
+            yield from get_paths_from_expression(func_expression)
+    if isinstance(expr, Case):
+        for case in expr.cases:
+            yield from get_children_from_q(case.condition)
+
+
 def get_children_from_q(q):
     for child in q.children:
         if isinstance(child, Node):
             yield from get_children_from_q(child)
         else:
-            yield child
+            if isinstance(child, tuple):
+                lhs, rhs = child
+                yield lhs
+                if hasattr(rhs, "resolve_expression"):
+                    yield from get_paths_from_expression(rhs)
+            if isinstance(child, ExpressionWrapper):
+                yield from get_children_from_q(child.expression)
```

Comment:
This is unnecessary with the simplified version of `get_paths_from_expression()` that handles `Q`:
```suggestion
```